File: simpleclaw_mobile/lib/core/utils/validators.dart
final RegExp _telegramTokenRegex = RegExp(r'^\d{6,15}:[A-Za-z0-9_-]{30,50}$');

bool isValidTelegramToken(String token) {
  return _telegramTokenRegex.hasMatch(token);
}

String getTelegramTokenError(String token) {
  if (token.isEmpty) {
    return '';
  }
  if (!isValidTelegramToken(token)) {
    return 'Неверный формат токена. Токен должен быть в формате: '
        '123456789:ABCdefGHIjklMNOpqrsTUVwxyz';
  }
  return '';
}

File: simpleclaw-rn/src/utils/validators.ts
const TELEGRAM_TOKEN_REGEX = /^\d{6,15}:[A-Za-z0-9_-]{30,50}$/;

export function isValidTelegramToken(token: string): boolean {
  return TELEGRAM_TOKEN_REGEX.test(token);
}

export function getTelegramTokenError(token: string): string {
  if (!token) return '';
  if (!isValidTelegramToken(token)) {
    return 'Неверный формат токена. Токен должен быть в формате: 123456789:ABCdefGHIjklMNOpqrsTUVwxyz';
  }
  return '';
}
